### PR TITLE
Add Rails 7.0 & Ruby 3.1 support

### PR DESCRIPTION
**Pull request by djpremier**
Fixes https://github.com/devise-security/devise-security/issues/352

- [x] Depends on https://github.com/blowmage/minitest-rails/pull/232 & https://github.com/blowmage/minitest-rails/issues/234
- [x] Depends on https://jira.mongodb.org/browse/RUBY-2834
- [x] Add ruby 3.1 to build matrix

MongoID add support to Rails 7.0 in version 7.3.4. See https://jira.mongodb.org/browse/MONGOID-5193.

Note that Rails 7.0.0 and Ruby 3.1 are incompatible, so version 7.0.1 is the minimum required. See https://github.com/rails/rails/issues/43998 and https://github.com/rails/rails/pull/43951

### CODE REVIEW

**Comment by djpremier** [comment]
Continuation of the work done by @olbrich.

PR https://github.com/devise-security/devise-security/pull/353

File: Appraisals [hunk 1]
```diff
@@ -24,3 +24,14 @@ appraise 'rails-6.1' do
...
   gem 'net-smtp', require: false
 end
+
+appraise 'rails-7.0' do
+  gem 'railties', '~> 7.0.1' # Rails 7.0.0 and Ruby 3.1 are incompatible. See https://github.com/rails/rails/issues/43998 and https://github.com/rails/rails/pull/43951
+  gem 'devise', '>= 4.8.1' # devise introduced Rails 7 support
+  gem 'minitest-rails', '~> 7.0.0'
+
+  # net-smtp, net-imap and net-pop were removed from default gems in Ruby 3.1, but is used by the `mail` gem.
+  # So we need to add them as dependencies until `mail` is fixed: https://github.com/mikel/mail/pull/1439
```
Comment (chipach) [inline Appraisals:34]: Totally pedantic note:  the referenced PR was closed in favor of https://github.com/mikel/mail/pull/1472 and mail [2.8.0.rc1](https://rubygems.org/gems/mail/versions/2.8.0.rc1) fixes this, but, alas, it is still a RC.  There's been a request to back port it to mail 2.7, but that hasn't happened.

But I would have done exactly what you did here.

**Comment by dillonwelch** [comment]
Thanks for working on this. My schedule has been pretty hectic so I hadn't been able to work on it, but I will do my best to review and get it merged & published!

**Comment by dillonwelch** [comment]
Happily surprised that it seems like no lib changes were needed to make this work.

I'm assuming that you're using this in a project? Would you be able to confirm that it works on Rails 7? Unfortunately the projects I work on now do not use this gem so it's hard for me to verify myself.

**Comment by chipach** [comment]
> Happily surprised that it seems like no lib changes were needed to make this work.
> 
> I'm assuming that you're using this in a project? Would you be able to confirm that it works on Rails 7? Unfortunately the projects I work on now do not use this gem so it's hard for me to verify myself.

We'd like to use it in our Rails 7 app and could report back if all goes well!

**Comment by djpremier** [comment]
> Happily surprised that it seems like no lib changes were needed to make this work.
> 
> I'm assuming that you're using this in a project? Would you be able to confirm that it works on Rails 7? Unfortunately the projects I work on now do not use this gem so it's hard for me to verify myself.

Hi @dillonwelch, yes I have been using it for about 2 months and without problems so far.

File: Appraisals (hunk 1 above)
Comment (djpremier) [reply Appraisals:34]: Yes, in this case I just replicated the comment that was for the other versions, but as soon as it is a stable release I will open a PR to adjust the minimum version of the mail.

**Review by olbrich — APPROVED** [review]

**Pull request merged**